runtime/envelope/envelope.go

```diff
@@ -153,19 +155,24 @@ func NewEnvelope(ctx context.Context, wlet *protos.EnvelopeInfo, config *protos.
 	}
 
 	// Make a temporary directory for unix domain sockets.
-	tmpDir, err := runtime.NewTempDir()
-	if err != nil {
-		return nil, err
-	}
-	runtime.OnExitSignal(func() { os.RemoveAll(tmpDir) }) // Cleanup when process exits
-
-	// Arrange to delete tmpDir if this function returns an error.
-	removeDir := true // Cleared on a successful return
-	defer func() {
-		if removeDir {
-			os.RemoveAll(tmpDir)
+	var removeDir bool
+	tmpDir := options.TmpDir
```

Comment:
If a user provides the `TmpDir` option, are we promising not to delete the directory? It seems like here, we don't delete it, but on line 257 in the `Serve` method, we do?

Reply:
Great catch. Thanks.